worldedit-core/src/main/java/com/boydti/fawe/beta/implementation/IChunkExtent.java

```diff
@@ -91,4 +105,88 @@ default int getOpacity(int x, int y, int z) {
         final IChunk chunk = getOrCreateChunk(x >> 4, z >> 4);
         return chunk.getOpacity(x & 15, y, z & 15);
     }
+
+    @Override
+    default Entity createEntity(Location location, BaseEntity entity) {
+        final IChunk chunk = getOrCreateChunk(location.getBlockX() >> 4, location.getBlockZ() >> 4);
+        Map<String, Tag> map = new HashMap<>(entity.getNbtData().getValue()); //do not modify original entity data
+        map.put("Id", new StringTag(entity.getType().getName()));
+        
+        //Set pos
+        List<DoubleTag> posList = new ArrayList<>();
+        posList.add(new DoubleTag(location.getX()));
+        posList.add(new DoubleTag(location.getY()));
+        posList.add(new DoubleTag(location.getZ()));
+        map.put("Pos", new ListTag(DoubleTag.class, posList));
+        
+        //set new uuid
+        UUID newuuid = UUID.randomUUID();
+        int[] uuidArray = new int[4];
+        uuidArray[0] = (int) (newuuid.getMostSignificantBits() >> 32);
+        uuidArray[1] = (int) newuuid.getMostSignificantBits();
+        uuidArray[2] = (int) (newuuid.getLeastSignificantBits() >> 32);
+        uuidArray[3] = (int) newuuid.getLeastSignificantBits();
+        map.put("UUID", new IntArrayTag(uuidArray));
+        
+        map.put("UUIDMost", new LongTag(newuuid.getMostSignificantBits()));
+        map.put("UUIDLeast", new LongTag(newuuid.getLeastSignificantBits()));
+        
+        map.put("PersistentIDMSB", new LongTag(newuuid.getMostSignificantBits()));
+        map.put("PersistentIDLSB", new LongTag(newuuid.getLeastSignificantBits()));
+        
+        chunk.setEntity(new CompoundTag(map));
+        return new IChunkEntity(this, location, newuuid, entity);
+    }
+
+    @Override
+    default void removeEntity(int x, int y, int z, UUID uuid) {
+        final IChunk chunk = getOrCreateChunk(x >> 4, z >> 4);
+        chunk.removeEntity(uuid);
+    }
+    
+    class IChunkEntity implements Entity {
```

Comment:
why are we adding this here?

Reply:
the EntityExtentCopy expects a non null result to remove entities (e.g. //cut, //move)